### PR TITLE
feat: Indirect prompt injection plugin

### PR DESCRIPTION
**Pull request by sklein12**

### CODE REVIEW

File: src/redteam/util.ts [hunk 1]
```diff
@@ -15,7 +15,8 @@ export async function retryWithDeduplication<T>(
   operation: (currentItems: T[]) => Promise<T[]>,
   targetCount: number,
   maxConsecutiveRetries: number = 2,
-  dedupFn: (items: T[]) => T[] = (items) => Array.from(new Set(items)),
+  dedupFn: (items: T[]) => T[] = (items) =>
+    Array.from(new Set(items.map((item) => JSON.stringify(item)))).map((item) => JSON.parse(item)),
```
Comment (mldangelo) [inline src/redteam/util.ts:19]: out of curiosity, why is this change necessary?

File: src/redteam/plugins/base.ts [hunk 2]
```diff
@@ -66,26 +68,38 @@ export abstract class PluginBase {
...
+      assert: this.getAssertions(prompt.prompt),
     }));
   }
 
+  protected parseGeneratedPrompts(generatedPrompts: string): { prompt: string }[] {
+    return generatedPrompts
+      .split('\n')
+      .filter((line: string) => line.includes('Prompt:'))
+      .map((line: string) => {
+        line = removeAsterisks(line);
```
Comment (mldangelo) [inline src/redteam/plugins/base.ts]: were you seeing this on a specific plugin or model?

File: src/commands/generate/redteam.ts [hunk 3]
```diff
@@ -107,8 +107,13 @@ export async function doGenerateRedteam(options: RedteamGenerateOptions) {
     typeof s === 'string' ? { id: s } : s,
   );
 
-  logger.debug(`plugins: ${plugins.map((p) => p.id).join(', ')}`);
-  logger.debug(`strategies: ${strategyObjs.map((s) => s.id ?? s).join(', ')}`);
+  try {
+    logger.debug(`plugins: ${plugins.map((p) => p.id).join(', ')}`);
+    logger.debug(`strategies: ${strategyObjs.map((s) => s.id ?? s).join(', ')}`);
+  } catch (error) {
+    logger.error('Error logging plugins and strategies. One did not have a valid id.');
```
Comment (mldangelo) [inline src/commands/generate/redteam.ts:112]: How do we get here? 

nit, consider other ways to handle this such as throwing and exiting or listing a message like `x, y, z plugins are invalid`

**Review by mldangelo — COMMENTED** [review]
First glance looks great, thank you! You should definitely add this to the guide as well.

File: src/redteam/util.ts (hunk 1 above)
Comment (sklein12) [reply src/redteam/util.ts:19]: generatePrompts now returns an array of objects instead of an array of strings, so the items will never be the same since they are just pointers to objects now instead of strings, we need to convert the objects into strings to see if they are dupes.

File: src/redteam/plugins/base.ts (hunk 2 above)
Comment (sklein12) [reply src/redteam/plugins/base.ts]: Whatever the default model is, I was seeing it on almost every single prompt response

File: src/commands/generate/redteam.ts (hunk 3 above)
Comment (sklein12) [reply src/commands/generate/redteam.ts:112]: You get this if you have a blank line 🤣
